conjure-java-jackson-serialization/src/main/java/com/palantir/conjure/java/serialization/InstrumentedJsonFactory.java

```diff
@@ -0,0 +1,710 @@
+
+        @Override
+        public String getText() throws IOException {
+            return recordStringLength(delegate.getText());
+        }
+
+        @Override
+        public int getText(Writer writer) throws IOException, UnsupportedOperationException {
+            return delegate.getText(writer);
+        }
+
+        @Override
+        public char[] getTextCharacters() throws IOException {
+            return delegate.getTextCharacters();
+        }
+
+        @Override
+        public int getTextLength() throws IOException {
+            return delegate.getTextLength();
+        }
+
+        @Override
+        public int getTextOffset() throws IOException {
+            return delegate.getTextOffset();
+        }
+
+        @Override
+        public boolean hasTextCharacters() {
+            return delegate.hasTextCharacters();
+        }
+
+        @Override
+        public Number getNumberValue() throws IOException {
+            return delegate.getNumberValue();
+        }
+
+        @Override
+        public Number getNumberValueExact() throws IOException {
+            return delegate.getNumberValueExact();
+        }
+
+        @Override
+        public NumberType getNumberType() throws IOException {
+            return delegate.getNumberType();
+        }
+
+        @Override
+        public byte getByteValue() throws IOException {
+            return delegate.getByteValue();
+        }
+
+        @Override
+        public short getShortValue() throws IOException {
+            return delegate.getShortValue();
+        }
+
+        @Override
+        public int getIntValue() throws IOException {
+            return delegate.getIntValue();
+        }
+
+        @Override
+        public long getLongValue() throws IOException {
+            return delegate.getLongValue();
+        }
+
+        @Override
+        public BigInteger getBigIntegerValue() throws IOException {
+            return delegate.getBigIntegerValue();
+        }
+
+        @Override
+        public float getFloatValue() throws IOException {
+            return delegate.getFloatValue();
+        }
+
+        @Override
+        public double getDoubleValue() throws IOException {
+            return delegate.getDoubleValue();
+        }
+
+        @Override
+        public BigDecimal getDecimalValue() throws IOException {
+            return delegate.getDecimalValue();
+        }
+
+        @Override
+        public boolean getBooleanValue() throws IOException {
+            return delegate.getBooleanValue();
+        }
+
+        @Override
+        public Object getEmbeddedObject() throws IOException {
+            return delegate.getEmbeddedObject();
+        }
+
+        @Override
+        public byte[] getBinaryValue(Base64Variant bv) throws IOException {
+            return delegate.getBinaryValue(bv);
+        }
+
+        @Override
+        public byte[] getBinaryValue() throws IOException {
+            return delegate.getBinaryValue();
+        }
```

Comment:
Do we need to track byte array lengths?

Reply:
I don't think this is enforced in Jackson. I agree that json array sizes (including but not limited to byte-arrays) should be bounded similarly to string length, but I'd prefer to build up to that later if that's alright with you.

Reply:
FWTW String value length limitations should apply for case of JSON (since physically they are JSON Strings).

But no limit checking for native Binary for Smile or CBOR etc. Could add limits in future, but if so those would be handled separately.

Reply:
Thanks for the heads up!

In most places these are used we have some form of protocol negotiation and expect smile and json to behave equivalently, so the data should still provide valid signal, but I will differentiate the metrics based on the format in a follow-up